### PR TITLE
Create new getting started docs for CMake.

### PR DESCRIPTION
**Pull request by ScottTodd**
Planning on writing similar docs for Bazel on each OS, then extending with instructions for Vulkan, Python bindings, and other leaf components.

### CODE REVIEW

File: docs/getting_started_linux_cmake.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Getting Started on Linux with CMake
+
+## Prerequisites
+
+### Install CMake
+
+Install CMake version >= 3.13:
+
+```shell
```
Comment (antiagainst) [inline docs/getting_started_linux_cmake.md:9]: Should this just be `sh`?

File: docs/getting_started_linux_cmake.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+repository, just make sure that they initialize the submodules.
+
+### Build
+
+Configure:
+
+```shell
+$ cmake -B build/ .
+```
+
```
Comment (antiagainst) [inline docs/getting_started_linux_cmake.md:66]: Put a tip here to link master `CMakeLists.txt` for more configuration controls?

File: docs/getting_started_linux_cmake.md (hunk 1 above)
Comment (ScottTodd) [reply docs/getting_started_linux_cmake.md:9]: `sh` is one of the aliases for `shell`: https://github.com/github/linguist/blob/40992ba7f86889f80dfed3ba95e11e1082200bad/lib/linguist/languages.yml#L4984-L5004 , so I think they are equivalent.

I've just been using `shell` consistently, but I don't have a strong opinion on which of the aliased names to use.

File: docs/getting_started_linux_cmake.md (hunk 2 above)
Comment (ScottTodd) [reply docs/getting_started_linux_cmake.md:76]: Good idea, done.

I'm trying to set a high bar for what instructions stay in this doc (I'm really on the fence about the "compatible" clang versions section...). I also added some comments to the top of these docs as guidance for others editing them in the future.

File: docs/getting_started_linux_cmake.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+Verify the version with `clang++ --version`. We have tested with the following
+versions:
+
+*   6.0.1
+*   8.0.1
+
+There have been issues with the following versions:
+
+*   [9.0.0-2](https://github.com/google/iree/issues/1216)
```
Comment (marbre) [inline docs/getting_started_linux_cmake.md:44]: The issue mentions **9.0.1**, not **9.0.0-2**. Anyway, neither had a problem on my Gentoo machine with 9.0.1 so far nor with the Ubuntu VM using 9.0.0-2...

Comment (ScottTodd) [reply docs/getting_started_linux_cmake.md]: I'll just remove the clang compatibility section then. It doesn't seem as useful if there's only one report of a version not working.

**Pull request closed without merging**